addon/index.js

```diff
@@ -1,4 +1,61 @@
 import moment from './lib';
+import Ember from 'ember';
+
+
+function hasOwnProp(a, b) {
+  return Object.prototype.hasOwnProperty.call(a, b);
+}
+
+function extend(a, b) {
+  for (let propName in b) {
+    if (hasOwnProp(b,propName)) {
+      Object.defineProperty(a,propName, {
+        enumerable: true,
+        configurable: true,
+        get() {
+          return b[propName];
+        },
+        set(newValue) {
+          b[propName] = newValue;
+        }
+      });
+    }
+  }
+
+  // non-enumerable property
+  ['toString', 'valueOf'].forEach(propName => {
+    if (hasOwnProp(b, propName)) {
+      Object.defineProperty(a, propName, {
+        enumerable: false,
+        configurable: true,
+        get() {
+          return b[propName];
+        },
+        set(newValue) {
+          b[propName] = newValue;
+        }
+      });
+    }
+  });
+
+  return a;
+}
+
+function createAugmentedMomentFrom(fn) {
+  const augmentedMoment = function() {
+    const m = fn(...arguments);
+    const meta = Ember.meta(m);
```

Comment:
Private API I believe, makes me slightly worried...

Reply:
Yes. I've not found how to do otherwise. The other solution is to use Ember.Object.

Reply:
Is this a show stopper?

Reply:
Yes.  I feel it will be painful to maintain.

Reply:
so we have to use `Ember.Object` to encapsulate Moment object...